plugin/documents.py

```diff
@@ -683,6 +686,7 @@ def _on_query_completions_async(self, resolve_completion_list: ResolveCompletion
         sessions = list(self.sessions_async('completionProvider'))
         if not sessions or not self.view.is_valid():
             resolve_completion_list([], 0)
+            self._auto_complete_triggered_manually = False  # reset state for next completion popup
```

Comment:
It's just a minor nit but for maintenance and clarity reasons I'd prefer that we unconditionally reset it right at the start of the method. Before that we can store it in a local and pass to `_on_all_settled`.

It's a suggestion and you can follow it if you agree that it would be better but feel free to ignore it if you want.